jscomp/core/js_pass_flatten_and_mark_dead.ml

```diff
@@ -39,35 +39,35 @@ type meta_info =
   | Recursive
 
 
+let super = Js_record_iter.iter
 
 let mark_dead_code (js : J.program) : J.program = 
   let ident_use_stats : meta_info Hash_ident.t
     = Hash_ident.create 17 in 
-  let mark_dead  : Js_iter.iter = object (self)
-    inherit Js_iter.iter
-    method! ident ident = 
+  let mark_dead  = { super with 
+    ident = (fun _ ident -> 
       (match Hash_ident.find_opt ident_use_stats ident with
        | None -> (* First time *)
          Hash_ident.add ident_use_stats ident Recursive 
        (* recursive identifiers *)
        | Some Recursive
          -> ()
-       | Some (Info x) ->  Js_op_util.update_used_stats x Used )
-    method! variable_declaration vd = 
+       | Some (Info x) ->  Js_op_util.update_used_stats x Used ));
+     variable_declaration = fun self vd -> 
       match vd.ident_info.used_stats with 
       |  Dead_pure 
         -> ()
       |  Dead_non_pure  -> 
         begin match vd.value with
           | None -> ()
-          | Some x -> self#expression x 
+          | Some x -> self.expression self x 
         end
```

Comment:
in theory, we can tie the knot here, and since mark_dead.expression is the same as super.expression, we can call super.expression directly
The downside is that if we do override super.expression, it will be wrong